crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -247,18 +247,24 @@ impl RealRibosome {
         dna_file: DnaFile,
         wasmer_module_cache: Arc<ModuleCacheLock>,
     ) -> RibosomeResult<Self> {
-        let shared_test_module_cache = temp_dir().join("deepkey_wasm_cache");
-        let _ = std::fs::create_dir_all(shared_test_module_cache.clone());
+        let mut _shared_test_module_cache: Option<PathBuf> = None;
+        #[cfg(test)]
+        {
+            // Create this temporary directory only in tests.
+            let shared_test_module_cache_dir = std::env::temp_dir().join("deepkey_wasm_cache");
+            let _ = std::fs::create_dir_all(shared_test_module_cache_dir.clone());
+            _shared_test_module_cache = Some(shared_test_module_cache_dir);
+        }
         let mut ribosome = Self {
             dna_file,
             zome_types: Default::default(),
             zome_dependencies: Default::default(),
             usage_meter: Self::standard_usage_meter(),
             wasmer_module_cache,
             #[cfg(test)]
-            shared_test_module_cache: Arc::new(ModuleCacheLock::new(ModuleCache::new(Some(
-                shared_test_module_cache,
-            )))),
+            shared_test_module_cache: Arc::new(ModuleCacheLock::new(ModuleCache::new(
+                _shared_test_module_cache,
+            ))),
```

Comment:
If there's a more elegant way of writing this, please fill me in!

Reply:
You could write a from to put the chained constructors elsewhere which might keep these lines tidier, but I think this code has to go somewhere :)